src/Controller/UserController.php

```diff
@@ -80,23 +81,57 @@ public function register(UserDto $userDto): Response
         return new JsonResponse($savedUserDto, Response::HTTP_CREATED);
     }
 
-//    /**
-//     * @Route(path="/{id}" methods={"DELETE"})
-//     */
-//    public function delete($id): Response
-//    {
-//        $user = $this->userRepository->findUserById($id);
-//
-//        if ($user) {
-//            $this->userRepository->remove($user);
-//            $this->logger->info('An user was deleted and removed from DB');
-//            return new JsonResponse();
-//        }
-//
-//
-//
-//        return new JsonResponse(['User with id' . $id . ' was deleted.'], Response::HTTP_OK);
-//    }
+    /**
+     * @Route(path="/{id}", methods={"DELETE"})
+     */
+    public function deleteUserAction($id): Response
+    {
+        $user = $this->userRepository->findUserById($id);
+
+        if (null === $user) {
+            return new JsonResponse('No user found', Response::HTTP_BAD_REQUEST);
+        }
+        $this->userRepository->remove($user);
+        $this->userRepository->add($user);
```

Comment:
why do you remove and then add the user? the idea is to remove a user with soft delete enabled

Reply:
solved.